PharmacyApi/Controllers/PharmacyController.cs

```diff
@@ -24,29 +24,34 @@ public class PharmacyController : ControllerBase
 
     #endregion
 
-    [HttpGet]
+    [HttpPost]
     [Route("all")]
-    public IActionResult GetAll() => Ok(_pharmacyService.GetAll());
-    
-    [HttpGet]
-    [Route("{id}")]
-    public async Task<IActionResult> GetById(int id)
+    [Route("{id?}")]
+    public async Task<IActionResult> GetPharmacy(int? id = null)
     {
-        var pharmacy = await _pharmacyService.GetById(id);
-
-        return pharmacy is not null
-            ? Ok(pharmacy)
-            : StatusCode(404, $"No pharmacy found with id {id}");
+        if (id is null)
+        {
+            var pharmacyListResult = await _pharmacyService.GetPharmacyListAsync();
+            return pharmacyListResult.Success 
+                ? Ok(pharmacyListResult.Result) 
+                : NotFound(pharmacyListResult.ErrorMessage);
+        }
+
+        var pharmacyResult = await _pharmacyService.GetByIdAsync(id.Value);
+        return pharmacyResult.Success 
+            ? Ok(pharmacyResult.Result) 
+            : NotFound(pharmacyResult.ErrorMessage);
```

Comment:
Why stop combining the calls at this level?
I see duplicate code... why not combine the operation on the Service layer?